### PR TITLE
transition period_helper to use pandas_datetimestruct

### PR DESCRIPTION
**Pull request by jbrockmendel**
In the last round we transitioned period.pyx to use pandas_datetimestruct instead of date_info.  This does the same thing for period_helper.

### CODE REVIEW

**Comment by jreback** [comment]
lgtm. ping on green.

**Comment by jreback** [comment]
actually, any doc-strings to update?

**Comment by jbrockmendel** [comment]
> actually, any doc-strings to update?

The affected docstrings are already updated.

File: pandas/_libs/tslibs/period.pyx [hunk 1]
```diff
@@ -635,15 +592,24 @@ def period_asfreq_arr(ndarray[int64_t] arr, int freq1, int freq2, bint end):
     return result
 
 
-def period_ordinal(int y, int m, int d, int h, int min,
-                   int s, int us, int ps, int freq):
-    return get_period_ordinal(y, m, d, h, min, s, us, ps, freq)
+cpdef int64_t period_ordinal(int y, int m, int d, int h, int min,
+                             int s, int us, int ps, int freq):
```
Comment (jreback) [inline pandas/_libs/tslibs/period.pyx]: doc-string

**Review by jreback — CHANGES_REQUESTED** [review]
doc-strings, can you rebase on master

**Comment by jreback** [comment]
you need to rebase on master

**Comment by jbrockmendel** [comment]
Ping

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**